tests/test_indexes_sharded.py

```diff
@@ -0,0 +1,40 @@
+# pylint: disable=missing-docstring,redefined-outer-name
+import time
+
+import pytest
+from pcsm import Runner
+from testing import Testing
+
+@pytest.mark.timeout(20)
+@pytest.mark.parametrize("phase", [Runner.Phase.CLONE])
+def test_inconsistent_index(t: Testing, phase: Runner.Phase):
+    t.source.admin.command({"enableSharding": "init_test_db"})
+    t.source.admin.command("shardCollection", "init_test_db.text_collection", key={"_id": "hashed"})
+    t.source["init_test_db"].text_collection.insert_many([{"a": {"b": i}, "words": f"text_{i}"} for i in range(50)])
+    t.source["init_test_db"].text_collection.insert_one({"a": {"b": []}, "words": "omnibus"})
+
+    # Create inconsistent index
+    try:
+        t.source["init_test_db"].text_collection.create_index([("a.b", 1), ("words", "text")])
+        assert False, "Index build should fail due array in doc for text index"
+    except Exception:
+        pass
```

Comment:
```suggestion
with pytest.raises(pymongo.errors.OperationFailure, match="text index"):
    t.source["init_test_db"].text_collection.create_index([("a.b", 1), ("words", "text")])
```

Something like this ought to work. The `assert False` is... strange IMO.

Reply:
Well it could be the first time seeing it, but it is common to write it like that in our QA e2e tests.